src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactory.java

```diff
@@ -154,39 +177,80 @@ private static double simulateRandomDraw(double mean, double deviation) {
 	/**
 	 * Call an API which generates a random address. This function aims at getting
 	 * the random address generated.
-	 *
-	 * @param retry indicates how many time the function has been called (Retry
-	 *              System). It will allow us to try escaping problems while
-	 *              contacting API (if it fails once, we try to call it again until
-	 *              the number specified in the class attribute NB_MAX_RETRY).
+	 * 
+	 * <p>
+	 * So, we generate a random latitude and a longitude and try to retrive an
+	 * address
+	 * <p>
+	 * 
+	 * <b>Regarding the API call:</b> <br>
+	 * <p>
+	 * After a multitude of tests, we have seen that after a failure of attitude and
+	 * length, we have a good one in any case <br>
+	 * <code>AddressApiException</code> should never be thrown. <br>
+	 * The probability of getting a good answer the first time is 1/5 on average
+	 * (after a multitude of trials)
```

Comment:
Bonjour, nous ne comprenons pas votre phrase _This method is designed so that its probability of failing to return (WITHOUT THROWING) is at most x_
En effet, cette fonction soulève une erreur s'il n'y a pas d'adresse. Il ne peut pas retourner une adresse vide car il soulèvera une exception 
cf ligne 249
`if (address.isPresent()) {
			return address.get();
		}`

Reply:
Bien vu, erreur de ma part : j’avais d’abord écrit dans l’autre sens (prob de succès sans throw) et puis j’ai inversé, mais imparfaitement. Il faut retirer "(without throwing)".

Reply:
Fix in 6a338d33ebf81607fa41f3766b6108320228291f